src/EnergyPlus/IndoorGreen.cc

```diff
@@ -0,0 +1,698 @@
+            }
+            // ZonePPFD
+            switch (ig.lightingMethod) {
+            case LightingMethod::LED: {
+                ig.ZPPFD = ScheduleManager::GetCurrentScheduleValue(state, ig.SchedLEDPtr) * ig.LEDNominalPPFD; // PPFD
+                ig.LEDActualPPFD = ig.LEDNominalPPFD;
+                ig.LEDActualEleP = ig.LEDNominalEleP;
+                ig.LEDActualEleCon = ig.LEDNominalEleP * Timestep;
+            } break;
+            case LightingMethod::Daylighting: {
+                ig.ZPPFD = 0;
+                ig.LEDActualPPFD = 0;
+                ig.LEDActualEleP = 0;
+                ig.LEDActualEleCon = 0;
+                if (!state.dataDayltg->CalcDayltghCoefficients_firstTime && state.dataEnvrn->SunIsUp) {
+                    ig.ZPPFD = state.dataDayltg->daylightControl(ig.LightControlPtr).refPts(1).lums[DataSurfaces::iLum_Illum] /
+                               77; // To be updated currently only take one reference point; 77 conversion factor from Lux to PPFD
+                }
+            } break;
+            case LightingMethod::LEDDaylighting: {
+                Real64 a = ScheduleManager::GetCurrentScheduleValue(state, ig.SchedLEDDaylightTargetPtr);
+                Real64 b = 0;
+                if (!state.dataDayltg->CalcDayltghCoefficients_firstTime && state.dataEnvrn->SunIsUp) {
+                    b = state.dataDayltg->daylightControl(ig.LightControlPtr).refPts(1).lums[DataSurfaces::iLum_Illum] /
+                        77; // To be updated currently only take one reference point; 77 conversion factor from Lux to PPFD
+                }
+                ig.LEDActualPPFD = max((a - b), 0.0);
+                if (ig.LEDActualPPFD >= ig.LEDNominalPPFD) {
+                    ig.ZPPFD = ig.LEDNominalPPFD + b; // LED Nominal + Daylight
+                    ig.LEDActualEleP = ig.LEDNominalEleP;
+                    ig.LEDActualEleCon = ig.LEDNominalEleP * Timestep;
+                } else {
+                    ig.ZPPFD = a; // Targeted PPFD
+                    ig.LEDActualEleP = ig.LEDNominalEleP * ig.LEDActualPPFD / ig.LEDNominalPPFD;
+                    ig.LEDActualEleCon = ig.LEDActualEleP * Timestep;
+                }
+            } break;
+            default:
+                break;
+            }
+            ZonePPFD = ig.ZPPFD;
+            ZoneVPD = ig.ZVPD / 1000; // kPa
+            // ET Calculation
+            switch (ig.etCalculationMethod) {
+            case ETCalculationMethod::PenmanMonteith: {
+                ig.ETRate = ETPenmanMonteith(state, ZonePreTemp, ZonePreHum, ZoneCO2, ZonePPFD, ZoneVPD, LAI);
+            } break;
+            case ETCalculationMethod::Stanghellini: {
+                ig.ETRate = ETStanghellini(state, ZonePreTemp, ZonePreHum, ZoneCO2, ZonePPFD, ZoneVPD, LAI);
+            } break;
+            case ETCalculationMethod::DataDriven: {
+                // set with EMS value if being called for.
+                if (ig.EMSETCalOverrideOn) {
```

Comment:
And actually -- you don't have to have a separate input option for actuated ET.  You can just default to one of the built-in options, and if the user is overriding it with an actuator, just use the actuated value.  This is what we do all over the code.  We can discuss this more for sure.

Reply:
It makes sense! I removed the input option for `data-driven`